### DISCUSSION TITLE
Deletion of records on the client without affecting the server

### DISCUSSION THREAD

**Discussion by maxrodro** [Mimetis/Dotmim.Sync#504] [Q&A]
Hi,

Is there some way in the DMS, to be able to delete records from some tables and these deleted ones are not synchronized to the server even if it has those tables configured as bidirectional

`   syncSetup.Tables["Employees"].SyncDirection = SyncDirection.Bidirectional;`

**Comment by Mimetis** [comment]
You can try something like
``` cs
agent.LocalOrchestrator.OnTableChangesSelected(args =>
{
    if (args.Changes == null || !args.Changes.HasRows)
        return;

    foreach(var changedRow in args.Changes.Rows.ToArray())
    {
        if (changedRow.RowState == DataRowState.Deleted)
            args.Changes.Rows.Remove(changedRow);
    }
});
```

**Reply by AbderrahmaneAhmam** [reply]
Hello @Mimetis 
In the current version this code doesn't work .

**Reply by Mimetis** [reply]
Indeed, the `OnTableChangesSelected` is happening later now, after rows are serialized.
You can use this kind of code now:

``` cs
agent.LocalOrchestrator.OnRowsChangesSelected(args =>
{
    if (args.SyncRow != null && args.SyncRow.RowState == SyncRowState.Deleted)
        args.SyncRow = null;
});

```

Let me know if it's working as expecting

**Reply by AbderrahmaneAhmam** [reply]
It's work great thanks.

I have a question, is there any news about advanced filter management as mentioned in this [Discussion](https://github.com/Mimetis/Dotmim.Sync/issues/255) is there any other solution to solve this problem instead of doing Reinitialize?

**Reply by Mimetis** [reply]
I tried to find a solution, but never succeed to fit with all scenarios.

**Reply by christoneethling** [reply]
Seeing as Advanced filter management is not in scope, perhaps its worth documenting the alternatives a bit more clearly and creating a sample app? I have create and example [here](https://github.com/Mimetis/Dotmim.Sync/discussions/1367#discussion-9169285) and once I know the correct\recommended approach I can help with the docs\sample if worthwhile
